browser/js/calculator.js

```diff
@@ -22,229 +28,247 @@
+  Calculator.prototype.setOutput = function ( text ) {
+    //FIXME: ghetto binding
+    var me = this
     if ( text ) {
-      if ( text.length > data.MAX_LENGTH ) {
-        shrinkFontSize()
+      if ( text.length > this.data.MAX_LENGTH ) {
+        me.shrinkFontSize()
       }
-      document.querySelector( elementSelectors.OUTPUT ).innerHTML = text
+      this.dom.querySelector( Calculator.elementSelectors.OUTPUT ).innerHTML = text
     } else {
-      if ( data.current.length > data.MAX_LENGTH ) {
-        shrinkFontSize()
+      if ( this.data.current.length > this.data.MAX_LENGTH ) {
+        me.shrinkFontSize()
       }
-      document.querySelector( elementSelectors.OUTPUT ).innerHTML = data.current
+      this.dom.querySelector( Calculator.elementSelectors.OUTPUT ).innerHTML = this.data.current
     }
   }
-  function setResult( number ) {
-    data.result = number.toString()
-    data.memory = data.result
+  Calculator.prototype.setResult = function ( number ) {
+    this.data.result = number.toString()
+    this.data.memory = this.data.result
   }
-  function clearResult() {
-    data.result = ''
+  Calculator.prototype
+  .clearResult = function () {
+    this.data.result = ''
   }
-  function resetData() {
-    data = {
+  Calculator.prototype
+  .resetData = function () {
+    this.data = {
       memory: '0',
       current: '0',
       operation: '',
       result: '',
       MAX_LENGTH: 11
     }
   }
-  function shrinkFontSize() {
-    document.querySelector( elementSelectors.OUTPUT )
-      .classList.add( cssClasses.SMALL_FONT )
+  Calculator.prototype.shrinkFontSize = function () {
+    this.dom.querySelector( Calculator.elementSelectors.OUTPUT )
+      .classList.add( Calculator.cssClasses.SMALL_FONT )
   }
-  function resetFontSize() {
-    document.querySelector( elementSelectors.OUTPUT )
-      .classList.remove( cssClasses.SMALL_FONT )
+  Calculator.prototype.resetFontSize = function () {
+    this.dom.querySelector( Calculator.elementSelectors.OUTPUT )
+      .classList.remove( Calculator.cssClasses.SMALL_FONT )
   }
-
-  function enterDigit( number ) {
-    var current = data.current
-    if ( current === '0' || data.memory === data.result ) {
-      clearResult()
+  Calculator.prototype.enterDigit = function ( number ) {
+    var me = this
+    var current = this.data.current
+    if ( current === '0' || this.data.memory === this.data.result ) {
+      me.clearResult()
       current = ''
     }
-    if ( current.length === data.MAX_LENGTH ) {
-      shrinkFontSize()
+    if ( this.data.current.length === this.data.MAX_LENGTH ) {
+      me.shrinkFontSize()
     }
     current += number
-    data.current = current
-    setOutput()
-  }
-  function setOperator( operation ) {
-    if ( data.memory === data.result ) {
-      data.memory = data.result
+    this.data.current = current
+    me.setOutput()
+  }
+  Calculator.prototype.setOperator = function ( operation ) {
+    //FIXME: ghetto binding
+    var me = this
```

Comment:
you dont need this here. `this` should work in all the `me` cases within the next 8 lines